libevmasm/Assembly.cpp

```diff
@@ -222,17 +222,21 @@ string Assembly::assemblyString(
 	return tmp.str();
 }
 
-Json::Value Assembly::createJsonValue(string _name, int _source, int _begin, int _end, string _value, string _jumpType)
+Json::Value Assembly::createJsonValue(string const& _name, AssemblyItem const& _item, std::string const& _value)
 {
 	Json::Value value{Json::objectValue};
 	value["name"] = _name;
-	value["source"] = _source;
-	value["begin"] = _begin;
-	value["end"] = _end;
+	value["begin"] = _item.location().start;
+	value["end"] = _item.location().end;
+	if (_item.m_modifierDepth != 0)
+		value["modifierDepth"] = static_cast<int>(_item.m_modifierDepth);
 	if (!_value.empty())
 		value["value"] = _value;
-	if (!_jumpType.empty())
-		value["jumpType"] = _jumpType;
+	if (!_item.getJumpTypeAsString().empty())
+	{
+		assertThrow(_name == "JUMP" && value.isMember("value") == false, AssemblyException, "");
+		value["value"] = _item.getJumpTypeAsString();
```

Comment:
Isn't this supposed to be `jumpType` rather than `value`? You have already filled `value` above and now you're overwriting it.
```suggestion
		value["jumpType"] = _item.getJumpTypeAsString();
```